### PR TITLE
Fixes Guardian Spirits not being able to move in stopped time along with their user.

### PR DESCRIPTION
**Pull request by Potato-Masher**
First PR, hooray for shit code. 

**I think this works, but I have no way of testing locally due to Guardian's two-player interaction.**

**EDIT: It did not work.**
**EDIT 2: It now does.**

:cl: Potato Masher
fix: The Wizard Federation has finally modified their production method for pre-packaged magic tarot cards for better compatibility with time-stopping spells. Guardians Spirits are no longer frozen by their user's time-stops.
/:cl:

6e59f0e0017f8ae6eed32d8507ef75b0

**Why:** This fix would make guardians be a little bit more in line with the whole "whatever affects the guardian affects the user" thing for consistency sake and make them just a little more useful.
On the topic of balance, this shouldn't be too bad. Just compare it to other options a wizard could deploy during a time-stop: 

>Disintegrate

Please note that attacking a time stop wizard with your own guardian (looking at you, miners) is still going to have both you and your guardian frozen as usual. You have to have the time stop spell learned to avoid it.

### CODE REVIEW

**Comment by Potato-Masher** [comment]
Erm. Well darn.

**Comment by Armhulen** [comment]
code/modules/fields/timestop.dm:31:error: summoner.mind.spell_list: undefined var

**Comment by Armhulen** [comment]
zoinks

**Comment by Potato-Masher** [comment]
Figures. That was the one I was eyeing. Not quite sure how to identify the summoner.

**Comment by Potato-Masher** [comment]
Erm, well, I'm not really sure how to locate the summoner with my level of knowledge. I'll wait for maintainer input. DNM for now.

File: code/modules/fields/timestop.dm [hunk 1]
```diff
@@ -27,6 +27,9 @@
 	for(var/mob/living/L in GLOB.player_list)
 		if(locate(/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop) in L.mind.spell_list) //People who can stop time are immune to its effects
 			immune[L] = TRUE
+	for(var/mob/living/simple_animal/hostile/guardian/G in GLOB.parasites)
+		if(locate(/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop) in var/mob/living/summoner.mind.spell_list) //It would only make sense that a person's stand would also be immune.
```
Comment (DaedalusGame) [inline code/modules/fields/timestop.dm]: first of all we need to check if our summoner even exists
`if(G.summoner && G.summoner.mind && locate(/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop) in G.summoner.mind.spell_list)`

**Comment by ExcessiveUseOfCobblestone** [comment]
Why don't you check to see if "L" has any soullinked mobs instead so you aren't having to locate the same thing twice.

File: code/modules/fields/timestop.dm (hunk 1 above)
Comment (Potato-Masher) [reply code/modules/fields/timestop.dm]: Thanks, I just realized this about 30 mins after you pointed it out.

**Comment by Potato-Masher** [comment]
Yee haw boys, all checks passed. Now we are getting somewhere. Thanks @DaedalusGame for the help.

File: code/modules/fields/timestop.dm [hunk 2]
```diff
@@ -27,6 +27,9 @@
 	for(var/mob/living/L in GLOB.player_list)
 		if(locate(/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop) in L.mind.spell_list) //People who can stop time are immune to its effects
 			immune[L] = TRUE
+	for(var/mob/living/simple_animal/hostile/guardian/G in GLOB.parasites)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/fields/timestop.dm]: is it not possible to do

 `for(var/mob/living/simple_animal/hostile/guardian/G in L.(soullinkedmobs var, I forgot what it's actually called)`
`>>>>immune[G] = TRUE`

without the need of double locating 🤔

**Comment by Kromgar** [comment]
I thought you couldn't attack anyone in time stop anyway? Other than firing fireballs

**Comment by Potato-Masher** [comment]
Erm, not sure. Time stop is honestly kind of a weird spell, and not everything is stopped by it (beepsky most noticeably). The least this will do is help consistency.

Now that I think about it, if you can't physically attack people (which is dumb), getting a ranged stand would bypass this.

File: code/modules/fields/timestop.dm (hunk 2 above)
Comment (Potato-Masher) [reply code/modules/fields/timestop.dm]: Are guardians soullinked? I'll hunt down this supposed var in a moment.

Comment (ExcessiveUseOfCobblestone) [reply code/modules/fields/timestop.dm]: They're the reason soul-linking exists iirc

**Comment by Kromgar** [comment]
Yeah actually ranged stand would be EXTREMELY useful in a timestop combo

**Comment by ExcessiveUseOfCobblestone** [comment]
Projectiles are stopped by timestop even if an immune mob shoots it (see wizard projectiles)

**Comment by ExcessiveUseOfCobblestone** [comment]
@RemieRichards TEACH ME PLEASE DONT JUST DOWNBOAT 😢

**Comment by Potato-Masher** [comment]
I have no clue myself on that matter, I see no evidence that says that guardians are on that list for some reason.  🤔

**Comment by RemieRichards** [comment]
@Potato-Masher @ExcessiveUseOfCobblestone 
Exactly, 1. they're not on the list, 2. nor are they the reason the list was created (I know, I made soul links).
Kor wrote guardians before that system, so guardian's code is almost copypaste now, it should be converted to a soul link, However, there's no hooks for soul links receiving/sharing damage, so you'd have to add that + a soul link type that uses it.

**Comment by ExcessiveUseOfCobblestone** [comment]
ill Look for the bridge today then and see if we can just do an easy change :^)

**Comment by Kromgar** [comment]
@ExcessiveUseOfCobblestone The point is is that the projectiles will be a fuckton harder to avoid

**Comment by ExcessiveUseOfCobblestone** [comment]
@Kromgar they could just melee you if they were any other stand and that has 100% reliability.

**Comment by Potato-Masher** [comment]
I wonder if a ranged stand loading up a ton of projectiles on one guy in timestop does more damage than a standard stand punching in timestop. Might be worth looking into I suppose in my spare _time_.

**Comment by ExcessiveUseOfCobblestone** [comment]
Does it matter when the wizard could have bough say disintegrate with those points and just lol-noped you?

**Comment by Potato-Masher** [comment]
Dammit, every discussion about wizard always goes back to disintegrate.
My only arguement for guardian is that it does not have a cooldown and can damage multiple people in a shorter amount of time, compared to disintegrate.

**Comment by Potato-Masher** [comment]
Also:

>tfw fucking Crab Rockets and steamed hams get merged faster than minor tweaks/fixes

**Comment by Potato-Masher** [comment]
Now that I think about it, this is more a fix than a tweak. Needs tag adjustment.

**Comment by Potato-Masher** [comment]
Whoops, did not mean to close.

**Pull request merged**